python/lsst/ip/isr/isrStatistics.py

```diff
@@ -340,81 +345,94 @@ def measureCti(self, inputExp, overscans, gains):
-                    else:
-                        values.append(osMean)
+        with gainContext(inputExp, image, applyGain, gains, isTrimmed=isTrimmed):
+            for ampIter, amp in enumerate(detector.getAmplifiers()):
+                ampStats = {}
+                readoutCorner = amp.getReadoutCorner()
+
+                # Full data region.
+                dataRegion = image[amp.getBBox() if isTrimmed else amp.getRawDataBBox()]
+
+                # Get the mean of the image
+                ampStats["IMAGE_MEAN"] = afwMath.makeStatistics(dataRegion, self.statType,
+                                                                self.statControl).getValue()
+
+                # First and last image columns.
+                pixelA = afwMath.makeStatistics(dataRegion.array[:, 0],
+                                                self.statType,
+                                                self.statControl).getValue()
+                pixelZ = afwMath.makeStatistics(dataRegion.array[:, -1],
+                                                self.statType,
+                                                self.statControl).getValue()
 
                 # We want these relative to the readout corner.  If that's
                 # on the right side, we need to swap them.
                 if readoutCorner in (ReadoutCorner.LR, ReadoutCorner.UR):
-                    ampStats["OVERSCAN_COLUMNS"] = list(reversed(columns))
-                    ampStats["OVERSCAN_VALUES"] = list(reversed(values))
+                    ampStats["FIRST_MEAN"] = pixelZ
+                    ampStats["LAST_MEAN"] = pixelA
+                else:
+                    ampStats["FIRST_MEAN"] = pixelA
+                    ampStats["LAST_MEAN"] = pixelZ
+
+                # Measure the columns of the overscan.
+                if overscans[ampIter] is None:
+                    # The amplifier is likely entirely bad, and needs to
+                    # be skipped.
+                    self.log.warning("No overscan information available for ISR statistics for amp %s.",
+                                     amp.getName())
+                    nCols = amp.getRawSerialOverscanBBox().getWidth()
+                    ampStats["OVERSCAN_COLUMNS"] = np.full((nCols, ), np.nan)
+                    ampStats["OVERSCAN_VALUES"] = np.full((nCols, ), np.nan)
                 else:
-                    ampStats["OVERSCAN_COLUMNS"] = columns
-                    ampStats["OVERSCAN_VALUES"] = values
+                    overscanImage = overscans[ampIter].overscanImage
+
+                    columns = []
+                    values = []
+                    for column in range(0, overscanImage.getWidth()):
+                        # If overscan.doParallelOverscan=True, the
+                        # overscanImage will contain both the serial
+                        # and parallel overscan regions.
+                        # Only the serial overscan correction is
```

Comment:
Should this be "serial CTI correction"?

Reply:
This is the measureCti part. I have another ticket open to also measure the parallel EPER here, so I'd rather keep this more general name rather than have to change the name, deprecate this method, and then ultimate deprecate it again later when I do add the parallel CTI calculation.

Reply:
Chris was referring to the middle word, not the first one.  "serial overscan correction" -> "serial CTI correction" ?